docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,220 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need a new api that will allow us to skip cluster/host validations, this will be for unsupported use.
+This API should work on a per cluster basis and will have it's own feature flag to allow us to enable it for specific users.
+An event should be emitted whenever the API is used during an installation.
+Clusters containing skipped validations should not contribute to statistics as these may be skewed.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- None identified at this stage
+
+## Proposal
+
+There are three main components as I see it
+
+### The REST API
```

Comment:
This seems very complex, both in the API and in the DB.  Why not just:
PUT api/v2/cluster/<id>/skip-host-validations (body has true/false)
GET api/v2/cluster/<id>/skip-host-validations (returns true/false)

Reply:
It will make the api simple but it will be a bit harder for us to understand the reason behind the failure. 
We will be able to to probably look at the events and understand from it what happened. 
but on the other side if they disable all the validations so we will loos basic validations that can actually 100 chance to fail.

I would not expect this api commonly use and it should be used only in extreme cases. 
So we can probably start simple and see how it goes, we can always add parameters to the api to improve it.

Reply:
So to clarify the proposal is to allow the skipping of host validations with a single setting and if this flag is set, we simply will not run any validations?

We can then put an item into a log to indicate we have done this, raise and event and then use the log entry / event to prevent the creation of triage tickets etc.